merkle-tree-system/main.go

```diff
@@ -71,6 +71,41 @@ func createAndStartAPI(tree merkletree.ExternalMerkleTree, port int) {
 	log.Fatal(http.ListenAndServe(":"+strconv.Itoa(port), router))
 }
 
+type tokenDataRequest struct {
+	Data string `json:"data"`
+}
+
+type tokenDataResponse struct {
+	MerkleAPIResponse
+	Token string `json:"token,omitempty"`
+}
+
+type MerkleAPIResponse struct {
+	Status bool   `json:"status"`
+	Error  string `json:"error,omitempty"`
+}
+
+func getToken() http.HandlerFunc {
```

Comment:
Why not take this method and the structures above into another file in the same package?

Reply:
logic related to the token generation & corresponding types/structures are separated in token.go